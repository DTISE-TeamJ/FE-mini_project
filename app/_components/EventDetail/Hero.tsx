"use client";

import React, { useState } from "react";
import Wrapper from "../Home/Events/Wrapper";
import Image from "next/image";
import Test from "@/assets/signupLandscape.webp";
import Calendar from "@/assets/icons/calendar.svg";
import Clock from "@/assets/icons/clock.svg";
import Ticket from "./Ticket";

const Hero = () => {
  const [isTicketVisible, setIsTicketVisible] = useState(false);

  const titleStyle = "text-xl font-medium";

  const handleClick = () => {
    setIsTicketVisible(true);
  };

  const handleCloseTicket = () => {
    setIsTicketVisible(false);
  };

  return (
    <>
      <div className="mt-20"></div>
      <Wrapper>
        <div className="flex justify-center items-center">
          <Image alt="" height={1000} src={Test} className="rounded-2xl" />
        </div>
        <div className="flex flex-col gap-12 my-10">
          <div className="flex justify-between items-start">
            <div className="text-5xl font-semibold">name</div>
            <div className="h-[160px] w-[320px] bg-white flex justify-center items-end shadow-xl rounded-2xl">
              <div></div>
              <button
                className="bg-cyan-500 text-white font-bold py-2 px-4 mb-4 rounded-full shadow-md hover:bg-blue-600 transition duration-300"
                onClick={handleClick}
              >
                Tickets
              </button>
            </div>
          </div>
          <div>
            <div className={titleStyle}>Organized by user.name</div>
          </div>
          <div>
            <div className={titleStyle}>Date and Time</div>
            <div className="flex items-start gap-2">
              <Image alt="" src={Calendar} width={25} />
              <span>date</span>
            </div>
            <div className="flex items-start gap-2">
              <Image alt="" src={Clock} width={25} />
              <span>start - end</span>
            </div>
          </div>
          <div>
            <div className={titleStyle}>Location</div>
            <div>location</div>
          </div>
          <div>
            <div className={titleStyle}>Event Description</div>
            <div>description</div>
          </div>
        </div>
      </Wrapper>
      {isTicketVisible && <Ticket onClose={handleCloseTicket} />}
    </>
  );
};

export default Hero;
