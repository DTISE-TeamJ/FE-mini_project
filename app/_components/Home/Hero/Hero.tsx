"use client";

import { useAppDispatch } from "@/store";
import { fetchEvents, searchEvents } from "@/store/action/event-slice";
import React, { useEffect, useState } from "react";
import { AiOutlineSearch } from "react-icons/ai";
import { useDebounce } from "use-debounce";

interface SearchProps {
  searchTerm: string;
  setSearchTerm: React.Dispatch<React.SetStateAction<string>>;
}

const Hero: React.FC<SearchProps> = ({ searchTerm, setSearchTerm }) => {
  const [debouncedSearchTerm] = useDebounce(searchTerm, 500);
  const dispatch = useAppDispatch();

  const handleSearchChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setSearchTerm(e.target.value);
  };

  useEffect(() => {
    if (debouncedSearchTerm) {
      dispatch(searchEvents(debouncedSearchTerm));
    } else {
      dispatch(fetchEvents());
    }
  }, [debouncedSearchTerm, dispatch]);

  return (
    <div className="w-full h-screen relative">
      <video
        className="w-full h-full object-cover"
        src="/beachVid.mp4"
        autoPlay
        muted
        loop
        preload="auto"
      />
      <div className="absolute w-full h-full top-0 left-0 bg-gray-900/30"></div>
      <div className="absolute top-0 w-full h-full flex flex-col justify-center text-center text-white p-4">
        <h1>First Class Travel</h1>
        <h2 className="py-4">Top 1% Locations Worldwide</h2>
        <form
          className="flex justify-between items-center max-w-[700px] mx-auto w-full border p-1
          rounded-md text-black bg-gray-100/90">
          <div>
            <input
              className="bg-transparent w-[300px] sm:w-[400px] font-[Poppins] focus:outline-none"
              type="text"
              placeholder="Search Destinations"
              value={searchTerm}
              onChange={handleSearchChange}
            />
          </div>
          <div>
            <AiOutlineSearch
              size={20}
              className="icon text-[#b9baba] cursor-pointer"
            />
          </div>
        </form>
      </div>
    </div>
  );
};

export default Hero;
