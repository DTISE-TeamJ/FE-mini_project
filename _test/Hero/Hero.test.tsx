import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import Hero from "app/_components/Home/Hero/Hero";

jest.mock("use-debounce", () => ({
  useDebounce: jest.fn((value) => [value]),
}));

describe("Hero Component", () => {
  const mockSetSearchTerm = jest.fn();

  beforeEach(() => {
    render(<Hero setSearchTerm={mockSetSearchTerm} />);
  });

  it("renders correctly and matches snapshot", () => {
    const { asFragment } = render(<Hero setSearchTerm={mockSetSearchTerm} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the main headings", () => {
    expect(screen.getByText("First Class Travel")).toBeInTheDocument();
    expect(screen.getByText("Top 1% Locations Worldwide")).toBeInTheDocument();
  });

  it("renders the search input", () => {
    expect(screen.getByPlaceholderText("Search Event")).toBeInTheDocument();
  });

  // it("renders the search icon", () => {
  //   expect(screen.getByRole("img", { hidden: true })).toBeInTheDocument();
  // });

  // it("renders the background video", () => {
  //   const video = screen.getByTestId("videoPlayer");

  //   expect(video).toBeInTheDocument();
  //   expect(video).toHaveAttribute("src", "/beachVid.mp4");
  //   expect(video).toHaveAttribute("autoplay");
  //   expect(video).toHaveAttribute("loop");
  //   expect(video).toHaveAttribute("muted");
  // });

  it("updates search input value when typed", () => {
    const input = screen.getByPlaceholderText("Search Event");
    fireEvent.change(input, { target: { value: "Beach vacation" } });
    expect(input).toHaveValue("Beach vacation");
  });

  it("calls setSearchTerm after debounce", async () => {
    const input = screen.getByPlaceholderText("Search Event");
    fireEvent.change(input, { target: { value: "Beach vacation" } });

    await new Promise((resolve) => setTimeout(resolve, 1000));

    expect(mockSetSearchTerm).toHaveBeenCalledWith("Beach vacation");
  });
});
