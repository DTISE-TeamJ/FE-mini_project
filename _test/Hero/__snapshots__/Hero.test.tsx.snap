// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Component renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-full h-screen relative"
  >
    <video
      autoplay=""
      class="w-full h-full object-cover"
      data-testid="videoPlayer"
      loop=""
      preload="auto"
      src="/beachVid.mp4"
    />
    <div
      class="absolute w-full h-full top-0 left-0 bg-gray-900/30"
    />
    <div
      class="absolute top-0 w-full h-full flex flex-col justify-center text-center text-white p-4"
    >
      <h1>
        First Class Travel
      </h1>
      <h2
        class="py-4"
      >
        Top 1% Locations Worldwide
      </h2>
      <form
        class="flex justify-between items-center max-w-[700px] mx-auto w-full border p-1 rounded-md text-black bg-gray-100/90"
      >
        <div>
          <input
            class="bg-transparent w-[300px] sm:w-[400px] font-[Poppins] focus:outline-none"
            placeholder="Search Event"
            type="text"
            value=""
          />
        </div>
        <div>
          <svg
            class="icon text-[#b9baba] cursor-pointer"
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
            />
          </svg>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
